src/components/Homepage/components/RestaurantList/RestaurantList.jsx

```diff
@@ -17,10 +19,26 @@ export const RestaurantList = () => {
     fetchData();
   }, []);
 
+  React.useEffect(() => {
+    setSearchValue(props.searchValue);
+  }, [props]);
```

Comment:
The dependency here should be `props.searchValue` instead of `props`, because otherwise this side effect will be called any time any prop changes. However, you need only `searchValue` as the dependency.

Reply:
ok, made changes